### ISSUE TITLE
Getting Timeout exception in IDE after 60secs even when 'newCommandTimeout' capability is given

### ISSUE THREAD

**Issue by Gocoollnath** [appium/appium-dotnet-driver#224]
## Description
Getting Timeout exception after 60secs in IDE,even when 'newCommandTimeout' capability is given. Server window waits for time given in the capability but the script doesn't run.

## Environment
Appium version 1.8.0
Desktop OS/version used to run Appium: Windows 10 Pro
Node.js version (unless using Appium.app|exe): 8.10.0
Mobile platform/version under test: Android API 25 ,v7.1.1
Real device or emulator/simulator: Real Device Samsung Tab
Appium CLI or Appium.app|exe: both
IDE: VisualStudio 2017 Enterprise and C# as coding language

Latest appium dot net driver from VS

## Code To Reproduce Issue [ Good To Have ]

DesiredCapabilities dc = new DesiredCapabilities();
        dc.SetCapability("deviceName", "Android device");
        dc.SetCapability("newCommandTimeout", 180);
       
        dc.SetCapability("app", "app path");
        
        dc.SetCapability("appWaitActivity", "*.Activity");
        dc.SetCapability("appWaitDuration", "40000");
        
        dc.SetCapability("autoGrantPermissions", true);
        driver = new AndroidDriver<AndroidElement>(new Uri("http://127.0.0.1:4723/wd/hub"), dc);

Login method code comes here.

## Ecxeption stacktraces

https://github.com/appium/appium/files/1987765/IDE.log

## Link to Appium logs

https://gist.github.com/Gocoollnath/08735d81ecce1ce068ba48c2a0cc89e0

**Comment by dpgraham** [comment]
According to your log, it's timing out after 180 seconds:

```
[warn] [35m[Appium][39m Closing session, cause was 'New Command Timeout of 180 seconds expired. Try customizing the timeout using the 'newCommandTimeout' desired capability'
```

**Comment by dpgraham** [comment]
What happens when you set it to `0` (`0` is infinite timeout)

**Comment by Gocoollnath** [comment]
Please see exception stack trace log. Server window times out after 180seconds but client after 60seconds

**Comment by Gocoollnath** [comment]
When i give 0 too, at times script timeouts after 60seconds

**Comment by dpgraham** [comment]
@TikhomirovSergey Do you have any ideas?

**Comment by vikrampvr** [comment]
Yes i am facing the same issue. Timing out after 60 seconds for  irrespective of the timout time mentioned.

![image](https://user-images.githubusercontent.com/31436539/40763306-29c01b62-64c2-11e8-8dfa-1fadfb2530eb.png)

**Comment by Gocoollnath** [comment]
@dpgraham @TikhomirovSergey  any updates? It is completely blocking our implementation

**Comment by Gocoollnath** [comment]
@dpgraham @TikhomirovSergey any updates? It is completely blocking our implementation

**Comment by githubdilip** [comment]
+one. I am also facing the issue after 60 even though i have give 180s. Any updates. Issue happens in parallel run.

**Comment by akinsolb** [comment]
duplicate of #221

**Comment by akinsolb** [comment]
See [this](https://github.com/appium/appium-dotnet-driver/issues/221#issuecomment-433962912) comment

**Comment by Gocoollnath** [comment]
Solved the issue using commanTimeout when driver is instantiated.

**Comment by JeevaTamil** [comment]
> What happens when you set it to `0` (`0` is infinite timeout)

worked for me
